### ISSUE TITLE
Mid-2012 MacBook Pro Doesn't See Battery without Magsafe Plugged In after Patch

### ISSUE THREAD

**Issue by Wildchemist** [BenSova/Patched-Sur#595]
**Hardware/Software**:
- Mid-2012 15" MacBook Pro, 2.7GHz, 16GB RAM, NVIDIA GeForce GT 650M w/1GB of GDDR5, 500MB SSD.
- Upgrading from: Catalina v10.15.7
- Upgrading to: Big Sur v11.4
- Latest version of Patched-Sur 0.1.2

**Issue**: Computer sees battery, charges battery, reports on battery health (it's a new battery, see clip from System Report, below), but when unplugged from charger laptop instantly does a hard shutdown. Didn't do this prior to the Patched Sur process. Furthermore, when shut down, if you unplug the computer from the Magsafe cable for any length of time, and then plug the cable back in, the computer will spontaneously start up as if you pressed the start button.

**To Reproduce**: Before installing your WiFi patch but after installing Patched Sur (running Big Sur with no WiFi) follow some other lame remedy steps for "Big Sur WiFi issues" such as what is found here: [https://osxdaily.com/2020/11/23/how-fix-macos-big-sur-wifi-issues/](https://osxdaily.com/2020/11/23/how-fix-macos-big-sur-wifi-issues/). Among the "fixes" are
1. reset the NVRAM and also 
2. reset the SMC by unplugging the battery, pressing the power switch while in powerless state for several seconds, then reboot.
3. No fix (of course)

Since none of this worked (and stupidly recognizing that the NVRAM reset also erased the previous WiFi patch...among other things?) I re-installed everything from scratch and applied the patches after step 15 using Terminal.
Big Sur now runs perfectly on my Mid-2012 MacBook Pro, and I have WiFi...But no battery, so I'm stuck running using the adapter until this is remedied.

**Expected behavior**
I expected that after I did the Patched Sur installation + patch that the computer would operate normally, that is enjoy Big Sur and access my WiFi.

**Screenshots**
Not a screenshot, but here is the System Information for "Power:"

> Battery Information:
> 
>   Model Information:
>   Manufacturer:	SMP
>   Device Name:	bq20z451
>   Pack Lot Code:	0
>   PCB Lot Code:	0
>   Firmware Version:	406
>   Hardware Revision:	2
>   Cell Revision:	165
>   Charge Information:
>   Fully Charged:	No
>   Charging:	No
>   Full Charge Capacity (mAh):	5823
>   State of Charge (%):	96
>   Health Information:
>   Cycle Count:	32
>   Condition:	Normal
> 
> System Power Settings:
> 
>   AC Power:
>   System Sleep Timer (Minutes):	10
>   Disk Sleep Timer (Minutes):	10
>   Display Sleep Timer (Minutes):	10
>   Wake on AC Change:	No
>   Wake on Clamshell Open:	Yes
>   Wake on LAN:	Yes
>   AutoPowerOff Delay:	259200
>   AutoPowerOff Enabled:	1
>   Current Power Source:	Yes
>   Display Sleep Uses Dim:	Yes
>   Hibernate Mode:	3
>   PrioritizeNetworkReachabilityOverSleep:	0
>   Battery Power:
>   System Sleep Timer (Minutes):	10
>   Disk Sleep Timer (Minutes):	10
>   Display Sleep Timer (Minutes):	2
>   Wake on AC Change:	No
>   Wake on Clamshell Open:	Yes
>   AutoPowerOff Delay:	259200
>   AutoPowerOff Enabled:	1
>   Display Sleep Uses Dim:	Yes
>   Hibernate Mode:	3
>   Reduce Brightness:	Yes
> 
> Hardware Configuration:
> 
>   UPS Installed:	No
> 
> AC Charger Information:
> 
>   Connected:	Yes
>   ID:	0x0100
>   Wattage (W):	85
>   Revision:	0x0000
>   Family:	0x0085
>   Serial Number:	0x00da45b1
>   Charging:	No
> 
> Power Events:
> 
> Next Scheduled Events:
> 
>   appPID:	877
>   Type:	Wake
>   Scheduled By:	sdbackupbytime
>   Time:	5/28/21, 9:00 PM
> 
>   appPID:	396
>   Type:	Wake
>   Scheduled By:	com.apple.alarm.user-visible-Weekly Usage Report
>   Time:	5/30/21, 10:28 AM
>   UserVisible:	0
> 

**Additional context**
So unlike the WiFi issue, where the WiFi hardware was not showing up in the System Report (but the Software Versions did show up), here the battery hardware shows up. I have a sinking suspicion that it was the SMC reset that severed the computer's proper connection with the battery. I believe the remedy will be re-installing Catalina and seeing if the problem goes away.... but if there is anything that sticks out I'm all ears. 

Many thanks! -Chris

**Comment by Kunolobo** [comment]
you write latest version of patch 0.12 but the latest one is 0.21 - try with this one.

**Comment by Wildchemist** [comment]
Apologies, the patch used was the April 25th version so 0.21.  A bit of late night lysdexia.

**Comment by BenSova** [comment]
This definitely sounds like a hardware problem. Patched Sur does absolutely nothing to your firmware, so your computer turning on when plugged in is crazy.

> reset the SMC by unplugging the battery, pressing the power switch while in powerless state for several seconds, then reboot.

That's the desktop way of resetting the SMC. You should really use [this method](https://support.apple.com/en-us/HT201295#reset) to do so. But since you mentioned that you did this and that your computer doesn't work without power, couldn't that mean the battery is not correctly plugged in?

> and stupidly recognizing that the NVRAM reset also erased the previous WiFi patch

The only "patch" affected by resetting NVRAM is the compatibility check patch (which is literally just `-no_compat_check` in your NVRAM `boot-args`).

**Comment by Wildchemist** [comment]
Hi Ben, many thanks for your efforts and contributions to the community!

I just tried the other SMC reset method, no luck. The instructions were a bit vague since the section with the keyboard combination says specifically "Notebook computers with a nonremovable battery" which mine is not so I looked farther down to the removable battery instructions.

I'll work with the laptop in its current state (I don't need a portable right now, necessarily) then convert it back to Catalina later to see if it helps address the issue.

I have to keep using Big Sur on it because my user account is located on an external SSD, currently plugged into an iMac Pro. This iMac Pro is being exchanged under AppleCare here at the end of its 3-year coverage because of an odd hardware issue so about a week to go through the mail-in and replace dance. Needs to be same OS version to best work with the external SSD user account and we have that, so mission accomplished irrespective of the power issue.

I'll check back in after the downgrade and let everyone know if it's indeed a hardware issue relating to the power if Catalina doesn't return the computer back to health. Although the battery is relatively new and reporting good health, it could still be a bad cell or something.

I'm also trying to get the Apple Hardware Test to work on an external drive, we'll see what that does.

Thanks again.

**Comment by BenSova** [comment]
Before anything, you checked that the battery was actually fully plugged back in right?

**Comment by Wildchemist** [comment]
> Before anything, you checked that the battery was actually fully plugged back in right?

Yes sir. Triple checked. Plus cleaned the connections gently with a toothpick, it is going on 10 years old! I'm still having a hard time getting an external drive to become bootable with the [AHT](https://github.com/upekkha/AppleHardwareTest), that will be a project for another day since it's happy hour now. Cheers!

**Comment by BenSova** [comment]
Like that page says, try `option d`, or if you installed recovery mode with the patcher, just `d` would work and give you Catalina Diagnostics.

**Comment by Wildchemist** [comment]
The D options have been generating this. It will remain with this error message and then reboot after about 3 or 4 minutes. Or who knows, maybe normally this model would shut down but mine just likes to be on whenever possible lol
![5753FB28-407A-4088-922B-175E034AC800_1_201_a](https://user-images.githubusercontent.com/83257218/120086040-1c335980-c0ab-11eb-8460-23c02b8186b7.jpeg)
